flask/app/datasets.py

```diff
@@ -340,12 +340,30 @@ def delete_dataset(id: int):
     dataset = (
         models.Dataset.query.filter(models.Dataset.dataset_id == id)
         .options(joinedload(models.Dataset.analyses))
+        .options(joinedload(models.Dataset.groups))
         .first_or_404()
     )
+
+    # Remove permission groups with the corresponding dataset ID in groups_datasets table
+    dataset.groups = []
+
+    tissue_sample = (
+        models.TissueSample.query.filter(
+            models.TissueSample.tissue_sample_id == dataset.tissue_sample_id
+        )
+        .options(joinedload(models.TissueSample.datasets))
+        .first_or_404()
+    )
```

Comment:
This should be retrieved by a joinedload on the dataset query. All the relationships are configured so the reverse side also exists, even if it's not in the class definition of the reverse side.